### PR TITLE
add registry secret integration tests

### LINKED ISSUE #652: Authenticated registries Integration tests
Integration tests for feature Support for fetching package repositories and packages from authenticated registry

### PR DESCRIPTION
**Pull request by maralavi**
**What this PR does / why we need it:**
- This PR is to add integration tests for imagepullsecret plugin
Fixes https://github.com/vmware-tanzu/tanzu-framework/issues/652
- Also made changes to make "imagepullsecret add" idempotent.
- changed --server-url flag for "tanzu secret registry add" to --server

**Describe testing done for PR:**
Added integration tests as part of the PR

**Does this PR introduce a user-facing change?:**
None

**Release note**:

```release-note
added integration tests for secret plugin, made secret plugin add operation idempotent and changed --server-url flag for "tanzu secret registry add" to --server
```

**New PR Checklist**

- [X] Ensure PR contains only public links or terms
- [X] Use good commit [messages](https://github.com/vmware-tanzu/tanzu-framework/blob/main/CONTRIBUTING.md)
- [X] Squash the commits in this branch before merge to preserve our git history
- [ ] If this PR is just an idea or POC, use a [Draft PR](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/about-pull-requests#draft-pull-requests) instead of a full PR
- [X] Add appropriate [kind label](../docs/release/kind-labels.md) according to what type of issue is being addressed.

### CODE REVIEW

**Comment by maralavi** [comment]
Integration tests would require the[ following PR ](https://github.com/vmware-tanzu/tanzu-framework/pull/832)to be merged to pass.

File: cmd/cli/plugin/secret/registry_secret_add.go [hunk 1]
```diff
@@ -4,47 +4,54 @@
...
 	"github.com/vmware-tanzu/tanzu-framework/pkg/v1/tkg/log"
 	"github.com/vmware-tanzu/tanzu-framework/pkg/v1/tkg/tkgpackageclient"
+	"github.com/vmware-tanzu/tanzu-framework/pkg/v1/tkg/tkgpackagedatamodel"
 )
 
 const errInvalidPasswordFlags = "exactly one of --password, --password-file, --password-env-var flags should be provided"
 
 var registrySecretAddCmd = &cobra.Command{
-	Use:   "add SECRET_NAME --server-url REGISTRY_URL --username USERNAME --password PASSWORD",
+	Use:   "add SECRET_NAME --server REGISTRY_SERVER --username USERNAME --password PASSWORD",
```
Comment (shivaani0505) [inline cmd/cli/plugin/secret/registry_secret_add.go:30]: @maralavi I see --server--url being changed to --server. Why do we need to do that

**Comment by shivaani0505** [comment]
@maralavi Changes LGTM. will approve once CAPD tests pass

**Review by shivaani0505 — APPROVED** [review]
LGTM

**Pull request merged**